hub/api/dataset.py

```diff
@@ -5,29 +5,62 @@
     InvalidKeyTypeError,
     UnsupportedTensorTypeError,
 )
-from typing import Union, Dict
+from hub.core.storage.provider import StorageProvider
+from hub.core.storage.memory import MemoryProvider
+from hub.core.storage.local import LocalProvider
+from hub.core.chunk_engine.read import read_dataset_meta, read_tensor_meta
+from hub.core.chunk_engine.write import write_array, write_dataset_meta
+from typing import Union, Dict, Optional
 import numpy as np
+import os
 
 
 class Dataset:
-    def __init__(self, path: str, mode: str = "a", ds_slice: slice = slice(None)):
+    def __init__(
+        self,
+        path: str,
+        mode: str = "a",
+        ds_slice: slice = slice(None),
```

Comment:
Not in the scope of this PR, but we might want to reconsider the slice. This works for datasets sliced by indexing like ds[5:7] but other cases such as dataset filtering may lead to non-contiguous indexes for eg. [0, 3, 8, 12]. We might want to address that too.

Reply:
We can build a wrapper class around slice that generates an index sequence if/when it comes to that.

Reply:
In the backlog :)